plugins/modules/ec2_spot_instance.py

```diff
@@ -0,0 +1,475 @@
+        device_name=dict(type='str'),
+        virtual_name=dict(type='str'),
+        ebs=dict(type='dict'),
+        no_device=dict(type='str'),
+        block_device_mappings=dict(type='list',elements='dict'),
+        network_interfaces=dict(type='list', elements='dict'),
+        associate_public_ip_address=dict(type='bool'),
+        delete_on_termination=dict(type='bool'),
+        description=dict(type='str'),
+        decive_index=dict(type='int'),
+        groups=dict(type='list'),
+        ipv6_address_count=dict(type='int'),
+        ipv6_addresses=dict(type=list, elements='dict'),
+        network_interface_id=dict(type='str'),
+        private_ip_address=dict(type='str'),
+        private_ip_addresses=dict(type='list', elements='dict'),
+        secondary_private_ip_address_count=dict(type='int'),
+        subnet_id=dict(type='str'),
+        associate_carrier_ip_address=dict(type='boolean'),
+        interface_type=dict(type='str', choices=['interface', 'efa']),
+        network_card_index=dict(type='int'),
+        ipv4_prefixes=dict(type='list', elements='dict'),
+        ipv4_prefix_count=dict(type='int'),
+        ipv6_prefixes=dict(type='list', elements='dict'),
+        ipv6_prefix_count=dict(type='int'),
+        spot_price=dict(type='str', default=''),
+        spot_type=dict(default='one-time', choices=["one-time", "persistent"]),
+        tags=dict(type='list', default=[]),
+        # valid_from=dict(type='datetime', default=datetime.datetime.now()),
+        # valid_until=dict(type='datetime', default=(datetime.datetime.now() + datetime.timedelta(minutes=60))
+        spot_instance_request_ids=dict(type='list', elements='str', default=[]),
+        placement=dict(type='dict'),
+        monitoring=dict(type='dict'),
+    )
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        check_boto3=False,
+        mutually_exclusive=[
+            ['block_duration', 'zone_group'],
+            ['block_duration', 'launch_group'],
+        ],
+        supports_check_mode=True
+    )
+
+    connection = module.client('ec2', AWSRetry.jittered_backoff())
+
+    state = module.params['state']
+
+    if state == 'present':
+        response = request_spot_instances(module, connection)
+        changed = True
+        spot_request = response.get('SpotInstanceRequests')[0]
+        module.exit_json(changed=changed, spot_request=spot_request)
+
+    if state == 'absent':
+        response = cancel_spot_instance_requests(module, connection)
+        changed = True
+        cancelled_spot_request_info = response.get('CancelledSpotInstanceRequests')
+        request_msg = ""
+        for each_item in cancelled_spot_request_info:
+            request_msg = request_msg + each_item.get('SpotInstanceRequestId') + ' '
+        request_msg = 'Spot requests with IDs: ' + request_msg + 'have been cancelled'
+        module.exit_json(changed=changed, cancelled_spot_request=request_msg)
```

Comment:
There request ID here should be the same as the parameter the user provided, so I would remove this logic and call 
`module.exit_json(changed=changed, msg='Cancelled Spot request {}'.format(module.params.get('spot_instance_request_ids'))` from inside the try block of `cancel_spot_instance_requests()`

changed status can be checked and set inside that function too.